libvast/src/system/node.cpp

```diff
@@ -332,14 +335,23 @@ node_state::spawn_command(const invocation& inv,
       VAST_DEBUG(this_node, "auto-generated new label:", label);
     }
   }
+  VAST_DEBUG(this_node, "spawns a", comp_name, "with the label", label);
+  auto spawn_inv = inv;
+  if (comp_name == "source") {
+    auto spawn_opt = caf::get<caf::settings>(spawn_inv.options, "spawn");
+    auto source_opt = caf::get<caf::settings>(spawn_opt, "source");
+    auto import_opt = caf::get<caf::settings>(spawn_inv.options, "import");
+    merge_settings(source_opt, import_opt);
+    put(spawn_inv.options, "import", import_opt);
```

Comment:
```suggestion
    spawn_inv.options["import"] = import_opt;
```

If this works, it feels way more idiomatic.